MetaMask/metamask-extension
# Transaction-list-item-details pop up to display the correct token information on token approve item

## Explanation
Fixes #17228 
Currently, for approve token pop up `Transaction-list-item-details` displays the details of only those tokens that are added to the wallet, if not a generic detail is displayed as shown in the ticket. With this fix, we are comparing the token address to the tokens available in detected tokens, tokens list, and custom tokens.



## Screenshots/Screencaps



### Before



### After



## Manual Testing Steps

1. Create a token from the test-dapp
2. Click approve token in the test-dapp
3. Once the token is approved, go to the activity tab and click Approve Token spending Cap and verify the amount in `spending cap`.



## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
